tools/serve/serve.py

```diff
@@ -925,6 +886,43 @@ def _get_paths(self, data):
         return rv
 
 
+def build_config(override_path=None, config_cls=ConfigBuilder, **kwargs):
```

Comment:
Moving the entire function makes it very difficult to see what has changed here. Please don't do that - if you're making changes to this function leave it on line 783 unless there's a specific reason not to.

Reply:
Due to some merging error in the previous commit, there were two build_config function. Apparently, I removed the wrong one

Reply:
I see, there was a reason the function was moved: We moved the wave logic from serve.py to wave.py and as @jgraham suggested, we pass the config builder class to the function, so we can have a different one defined in the wave.py. For this to not brake other code we added the default value of the local ConfigBuilder to the paramter, which is only accessible after its declaration. If you have another suggestion for this, please let me know.

Reply:
Ack, that sounds like a specific reason to keep this to me :). Thanks for the answer!